pkg/sql/plan2/function/builtin/unary/asin.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unary
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/asin"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"golang.org/x/exp/constraints"
+)
+
+func Asin[T constraints.Integer | constraints.Float](vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
```

Comment:
just type float64.  No other type

Reply:
just reserve float64. Remove the T

Reply:
If I remove the generic type `T` here, and replace the `origVecCol := vector.MustTCols[T](origVec)` into `origVecCol := vector.MustTCols[float64](origVec)`, the [unit test](https://github.com/matrixorigin/matrixone/pull/3586/files#diff-ad84ba0ea1ee2d9e5a31e3d618a6c3fd2c9f24b920d2219c0f553cb588bf2489R28) will fail because input vector types in unit test vary from int to float.

Reply:
So need I remove [unit test](https://github.com/matrixorigin/matrixone/pull/3586/files#diff-ad84ba0ea1ee2d9e5a31e3d618a6c3fd2c9f24b920d2219c0f553cb588bf2489R28) here? Further more, what's design purpose here? It seems that every numerical value will convert to `float64` before calling `asin`.

Reply:
Most math functions only accept float64 type in 0.6.